appbuilder/providers/device-app-data-provider.ts

```diff
@@ -0,0 +1,339 @@
+const DEVICE_TMP_DIR_FORMAT_V3 = "/mnt/sdcard/Android/data/%s/files/12590FAA-5EDD-4B12-856D-F52A0A1599F2";
+const CHECK_LIVESYNC_INTENT_NAME = "com.telerik.IsLiveSyncSupported";
+const IOS_PROJECT_PATH = "/Library/Application Support/12590FAA-5EDD-4B12-856D-F52A0A1599F2";
+const IOS_NS_PROJECT_PATH = "/tmp/Application Support/LiveSync";
+
+export class AndroidAppIdentifier extends DeviceAppDataBase implements ILiveSyncDeviceAppData {
+	private _deviceProjectRootPath: string = null;
+	private _liveSyncVersion: number;
+
+	constructor(_appIdentifier: string,
+		public device: Mobile.IDevice,
+		public platform: string,
+		private $errors: IErrors,
+		private $deployHelper: IDeployHelper,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants) {
+		super(_appIdentifier);
+	}
+
+	public get deviceProjectRootPath(): string {
+		if (!this._deviceProjectRootPath) {
+			let deviceTmpDirFormat = "";
+
+			let version = this.getLiveSyncVersion().wait();
+			if (version === 2) {
+				deviceTmpDirFormat = DEVICE_TMP_DIR_FORMAT_V2;
+			} else if (version === 3) {
+				deviceTmpDirFormat = DEVICE_TMP_DIR_FORMAT_V3;
+			} else {
+				this.$errors.failWithoutHelp(`Unsupported LiveSync version: ${version}`);
+			}
+
+			this._deviceProjectRootPath = this.getDeviceProjectRootPath(util.format(deviceTmpDirFormat, this.appIdentifier));
+		}
+
+		return this._deviceProjectRootPath;
+	}
+
+	public get liveSyncFormat(): string {
+		return null;
+	}
+
+	public encodeLiveSyncHostUri(hostUri: string): string {
+		return hostUri;
+	}
+
+	public getLiveSyncNotSupportedError(): string {
+		return `You can't LiveSync on device with id ${this.device.deviceInfo.identifier}! Deploy the app with LiveSync enabled and wait for the initial start up before LiveSyncing.`;
+	}
+
+	public isLiveSyncSupported(): IFuture<boolean> {
+		return (() => {
+			let isApplicationInstalled = this.device.applicationManager.isApplicationInstalled(this.appIdentifier).wait();
+			if (!isApplicationInstalled) {
+				this.$deployHelper.deploy(this.$devicePlatformsConstants.Android.toLowerCase()).wait();
+			}
+		 	return this.getLiveSyncVersion().wait() !== 0;
+		}).future<boolean>()();
+	}
+
+	private getLiveSyncVersion(): IFuture<number> {
+		return (() => {
+			if (!this._liveSyncVersion) {
+				this._liveSyncVersion = (<Mobile.IAndroidDevice>this.device).adb.sendBroadcastToDevice(CHECK_LIVESYNC_INTENT_NAME, {"app-id": this.appIdentifier}).wait();
+			}
+			return this._liveSyncVersion;
+		}).future<number>()();
+	}
+}
+
+export class AndroidCompanionAppIdentifier extends DeviceAppDataBase implements ILiveSyncDeviceAppData {
+	constructor(_appIdentifier: string,
+		public device: Mobile.IDevice,
+		public platform: string,
+		private $companionAppsService: ICompanionAppsService,
+		private $projectConstants: IProjectConstants) {
+		super($companionAppsService.getCompanionAppIdentifier($projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova, platform));
```

Comment:
According to the rest of the code, you can just call `super(_appIdentifier)` here